sklearn/ensemble/tests/test_stacking.py

```diff
@@ -0,0 +1,419 @@
+    "stacking_estimator",
+    [StackingClassifier(estimators=[('lr', LogisticRegression()),
+                                    ('rf', RandomForestClassifier()),
+                                    ('svm', LinearSVC())]),
+     StackingRegressor(estimators=[('lr', LinearRegression()),
+                                   ('rf', RandomForestRegressor()),
+                                   ('svm', LinearSVR(max_iter=1e4))])]
+)
+def test_stacking_named_estimators_dropped(stacking_estimator):
+    stacking_estimator.set_params(rf='drop')
+    stacking_estimator.fit(scale(X_iris), y_iris)
+    estimators = stacking_estimator.named_estimators_
+    assert 'rf' not in estimators.keys()
+    assert len(estimators) == 2
+    assert sorted(list(estimators.keys())) == sorted(['lr', 'svm'])
+
+
+@pytest.mark.parametrize(
+    "stacking_estimator",
+    [StackingClassifier(estimators=[('lr', LogisticRegression()),
+                                    ('svm', LinearSVC())]),
+     StackingRegressor(estimators=[('lr', LinearRegression()),
+                                   ('svm', LinearSVR())])]
+)
+def test_stacking_set_get_params(stacking_estimator):
+    params = stacking_estimator.get_params()
+    assert 'lr' in list(params.keys())
+    assert 'svm' in list(params.keys())
+
+    stacking_estimator.set_params(lr='drop')
+    params = stacking_estimator.get_params()
+    assert params['lr'] == 'drop'
+
+
+@pytest.mark.parametrize(
+    "estimator, X, y",
+    [(StackingClassifier(
+        estimators=[('lr', LogisticRegression(random_state=0)),
+                    ('svm', LinearSVC(random_state=0))]),
+      X_iris[:100], y_iris[:100]),  # keep only classes 0 and 1
+     (StackingRegressor(
+         estimators=[('lr', LinearRegression()),
+                     ('svm', LinearSVR(random_state=0))]),
+      X_diabetes, y_diabetes)],
+    ids=['StackingClassifier', 'StackingRegressor']
+)
+def test_stacking_randomness(estimator, X, y):
+    estimator_full = clone(estimator)
+    estimator_full.set_params(
+        cv=KFold(shuffle=True, random_state=np.random.RandomState(0))
+    )
+
+    estimator_drop = clone(estimator)
+    estimator_drop.set_params(lr='drop')
+    estimator_drop.set_params(
+        cv=KFold(shuffle=True, random_state=np.random.RandomState(0))
+    )
+
+    assert_allclose(
+        estimator_full.fit(X, y).transform(X)[:, 1:],
+        estimator_drop.fit(X, y).transform(X)
+    )
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    [StackingClassifier(
+        estimators=[('lr', LogisticRegression(random_state=0)),
+                    ('tree', DecisionTreeClassifier(random_state=0))]),
+     StackingRegressor(
+         estimators=[('lr', LinearRegression()),
+                     ('tree', DecisionTreeRegressor(random_state=0))])],
+    ids=['StackingClassifier', 'StackingRegressor']
+)
+def test_check_estimators_stacking_estimator(estimator):
+    # FIXME: to be removed when meta-estimators can specified themselves
```

Comment:
I don't think this deserves to be a FIXME. This may be true of every use of `check_estimator` in sklearn.